src/spanningtrees/boruvka.jl

```diff
@@ -0,0 +1,73 @@
+"""
+boruvka_mst(g, distmx = weights(g); minimize = true)
+Return a tuple `(mst, weights)` where `mst` is a vector of edges representing the
+optimum (minimum, by default) spanning tree of a connected, undirected graph
+`g` with optional matrix `distmx` that provides distinct edge weights, and
+`weights` is the sum of all the edges in the solution by using
+[Boruvka's algorithm](https://en.wikipedia.org/wiki/Bor%C5%AFvka%27s_algorithm).
+The algorithm correctly gives a minimum/maximum spanning tree if all edges have
```

Comment:
I don't understand this part, do some other algorithms not give the correct result for different weights? Or does the algorithm require weights to be different?

Reply:
The algorithm require different weights

Reply:
Maybe write it explicitly then. This algorithm requires weights to be non-identical.

Reply:
oh, I`ve wrote it. "if all edges have different weights", it is on the next line

Reply:
Yes I saw it, but it's the way the sentence is constructed which may be confusing:
` The algorithm correctly gives a minimum/maximum spanning tree if all edges have different weights`
==> the statement does not tell you you should not use the algorithm when it is not the case

Reply:
oooh okay, I'm sorry. I'm going to change it as soon as possible

Reply:
Do you think  "The algorithm correctly... only if" is better?

Reply:
"requires" would communicate better the meaning here IMO

Reply:
I think I've solved it, take a look and give me the feedback, please

Reply:
I have a second doubt (sorry for this pickiness):
"all edges have different weights":
so [1.5, 1.5, 1.5] is not correct, [1.0, 2.0, 3.0] is correct.
What about `[1.0, 2.0, 3.0]`?
Requires, that all weights are different => this would be invalid
Requires, that not all weights are identical => this would be valid

Reply:
I don`t if I understood what you really meant but I think I do, It requires all weights to be different so [1.0, 2.0, 3.0] is correct,  so [1.0, 2.0, 2.0] is wrong and  [1.0, 1.0, 1.0] is also wrong.

Reply:
ok perfect that's what I understood :)